### PR TITLE
eTags should be Weak eTags

### PR DESCRIPTION
**Pull request by Rarian**
This is a new pull request to address outstanding comments on #920.

### CODE REVIEW

**Comment by awoods** [comment]
This PR does not compile:
https://github.com/fcrepo4/fcrepo4/commit/71c568cf19c8589540f041619e3f50b08d070c80

There will also be a checkstyle error (line too long):
https://github.com/fcrepo4/fcrepo4/pull/991/files#diff-d372035f89c8d0846920e20bd13e0145R483

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java [hunk 1]
```diff
@@ -476,11 +476,11 @@ protected static void addCacheControlHeaders(final HttpServletResponse servletRe
 
         final FedoraResource mutableResource = resource instanceof NonRdfSourceDescription
                 ? ((NonRdfSourceDescription) resource).getDescribedResource() : resource;
-        final EntityTag etag = new EntityTag(mutableResource.getEtagValue());
+        final EntityTag etag = new EntityTag(mutableResource.getEtagValue(), true);
         final Date date = mutableResource.getLastModifiedDate();
 
         if (!etag.getValue().isEmpty()) {
-            servletResponse.addHeader("ETag", etag.toString());
+            servletResponse.addHeader("ETag", mutableResource.hasType(FEDORA_NON_RDF_SOURCE_DESCRIPTION) ? "" : "W/" ) + etag);
```
Comment (awoods) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: The inclusion of `W/` should be handled by the `EntityTag` class, not inserted manually here.
Why exactly are you suggesting that non-rdf-source-descriptions not have weak etags?

Comment (Rarian) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: I was blithely folllowing @ajs6f's comment here https://github.com/fcrepo4/fcrepo4/pull/920#discussion_r47252727

Comment (awoods) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: Fortunately, `EntityTag` handles those details for you.

Comment (Rarian) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: I just reread your comment (https://github.com/fcrepo4/fcrepo4/pull/920#issuecomment-151646374) and discovered that I had stashed some changes when I last worked onthis.  Will sort this out in a bit.

**Comment by awoods** [comment]
Superseded by: https://github.com/fcrepo4/fcrepo4/pull/1040

**Pull request closed without merging**